homeassistant/components/media_player/directv.py

```diff
@@ -47,73 +55,287 @@
+        vol.All(cv.time_period, cv.positive_timedelta)
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the DirecTV platform."""
     known_devices = hass.data.get(DATA_DIRECTV, set())
-    hosts = []
+    directv_entity = None
 
     if CONF_HOST in config:
+        discovered = False
+        name = config.get(CONF_NAME)
+        host = config.get(CONF_HOST)
+        port = config.get(CONF_PORT)
+        device = config.get(CONF_DEVICE)
         _LOGGER.debug("Adding configured device %s with client address %s ",
-                      config.get(CONF_NAME), config.get(CONF_DEVICE))
-        hosts.append([
-            config.get(CONF_NAME), config.get(CONF_HOST),
-            config.get(CONF_PORT), config.get(CONF_DEVICE)
-        ])
+                      name, device)
+        from DirectPy import DIRECTV
+        try:
+            dtv = DIRECTV(host, port, device)
+        except requests.exceptions.RequestException as ex:
+            # Use uPnP data only
+            _LOGGER.debug("Request exception %s trying to connect to %s",
+                          ex, host)
+            resp = []
+
+        if dtv:
+            try:
+                resp = dtv.get_version()
+            except requests.exceptions.RequestException as ex:
+                # Use uPnP data only
+                _LOGGER.debug("Request exception %s trying to get "
+                              "receiver id for %s", ex, name)
+                resp = []
+
+        client_discover_interval = config.get(CONF_CLIENT_DISCOVER_INTERVAL,
+                                              DEFAULT_CLIENT_DISCOVER_INTERVAL)
+        if client_discover_interval < timedelta(seconds=5):
+            _LOGGER.warning("Invalid client_discover_interval value %s "
+                            "provided for %s, minimum is 5 seconds",
+                            client_discover_interval, name)
+            client_discover_interval = DEFAULT_CLIENT_DISCOVER_INTERVAL
+
+        directv_entity = {
+            CONF_NAME: name,
+            CONF_HOST: host,
+            CONF_PORT: port,
+            CONF_DEVICE: device,
+            RECEIVER_ID: resp.get('receiverId', host).replace(' ', ''),
+            CONF_DISCOVER_CLIENTS: config.get(CONF_DISCOVER_CLIENTS),
+            CONF_CLIENT_DISCOVER_INTERVAL: client_discover_interval,
+        }
 
     elif discovery_info:
+        discovered = True
         host = discovery_info.get('host')
-        name = 'DirecTV_{}'.format(discovery_info.get('serial', ''))
-
-        # Attempt to discover additional RVU units
-        _LOGGER.debug("Doing discovery of DirecTV devices on %s", host)
+        serial_num = discovery_info.get('serial', host)
+        receiver_id = discovery_info.get('serial', '').split('-')[-1]
+        receiver_id = receiver_id if receiver_id else host
+        name = 'DirecTV_{}'.format(serial_num)
+        resp = []
+
+        if (receiver_id, DEFAULT_DEVICE) in known_devices:
```

Comment:
This will search if the tuple `(receiver_id, DEFAULT_DEVICE)` exists inside `known_devices`, is that what you want?

Reply:
Yes, that is exactly what I want. I'm checking if that receiver was already added as an entity. A receiver is unique through its combination of receiver_id (serial number) and device ID.